packages/core/src/routes/interaction/index.ts

```diff
@@ -48,6 +51,42 @@ export default function interactionRoutes<T extends AnonymousRouter>(
     }
   );
 
+  router.patch(
+    identifierPrefix,
+    koaInteractionBodyGuard(),
+    koaSessionSignInExperienceGuard(provider),
+    async (ctx, next) => {
+      const { event } = ctx.interactionPayload;
+      const interactionStorage = await getInteractionStorage(ctx, provider);
+
+      // Forgot Password specific event interaction can't be shared with other types of interactions
+      assertThat(
+        event === Event.ForgotPassword
+          ? interactionStorage.event === Event.ForgotPassword
+          : interactionStorage.event !== Event.ForgotPassword,
+        new RequestError({ code: 'session.verification_session_not_found' })
+      );
+
+      const identifierVerifiedInteraction = await identifierVerification(
+        ctx,
+        provider,
+        interactionStorage
+      );
+
+      const interaction = await profileVerification(ctx, provider, identifierVerifiedInteraction);
+
+      if (event !== Event.ForgotPassword) {
+        await mandatoryUserProfileValidation(ctx, interaction);
```

Comment:
nit: Still, we need to rename these functions with `verb-object` expressions

Reply:
renamed all method in the following PR.